pkg/config/network.go

```diff
@@ -58,3 +61,28 @@ func getClusterDNS(serviceCIDR string) (string, error) {
 
 	return dnsClusterIP.String(), nil
 }
+
+func (c *Config) EnsureNetworksDontOverlap() error {
```

Comment:
This could use a more descriptive name. From this name, I can't tell which networks should not overlap with which other networks.